File: src/main/java/com/adamkoch/iptables/TimeSchedule.java
package com.adamkoch.iptables;

import com.adamkoch.annotations.Unstable;
import java.util.ArrayList;
import java.util.List;
import java.util.StringJoiner;

/**
 *
 * @since 0.1.0
 * @author aakoch
 */
@Unstable
public class TimeSchedule {

  private final List<TimeRange> timeRanges;

  public TimeSchedule() {
    timeRanges = new ArrayList<>();
  }

  public void add(final TimeRange daySchedule) {
    timeRanges.add(daySchedule);
  }

  public List<TimeRange> getTimeRanges() {
    return timeRanges;
  }

  @Override
  public String toString() {
    StringJoiner sj = new StringJoiner(", ");
    timeRanges.forEach(daySchedule -> {
      sj.add(daySchedule.toString());
    });
    return sj.toString();
  }
}

File: src/main/java/com/adamkoch/iptables/TimeSchedule.kt
package com.adamkoch.iptables

import com.adamkoch.annotations.Unstable
import java.util.*
import java.util.function.Consumer

/**
 *
 * @since 0.1.0
 * @author aakoch
 */
@Unstable
class TimeSchedule {
    private val timeRanges: MutableList<TimeRange>
    fun add(daySchedule: TimeRange) {
        timeRanges.add(daySchedule)
    }

    fun getTimeRanges(): List<TimeRange> {
        return timeRanges
    }

    override fun toString(): String {
        val sj = StringJoiner(", ")
        timeRanges.forEach(Consumer { daySchedule: TimeRange -> sj.add(daySchedule.toString()) })
        return sj.toString()
    }

    init {
        timeRanges = ArrayList()
    }
}